### PR TITLE
Reflection_Engine: Added TryRunExtensionMethod

### LINKED ISSUE #2348: Reflection_Engine: introduce TryRunExtensionMethod
#### Description:

As discussed in https://github.com/BHoM/BHoM_Engine/pull/2343#pullrequestreview-597742860.

### PR DESCRIPTION
**Pull request by alelom**
### Issues addressed by this PR

Closes #2348

Adds a TryRunExtensionMethod. If no method is found, returns false. Otherwise, invokes the method and passes the output in the `out` parameter, then returns true.

### Test files

Test with any choice of extension methods.

### CODE REVIEW

File: Reflection_Engine/Compute/TryRunExtensionMethod.cs
Comment (pawelbaran) [inline Reflection_Engine/Compute/TryRunExtensionMethod.cs:50]: Successful invocation is not a condition driving the output - the output is equal only to whether the method was found or not.

File: Reflection_Engine/Compute/TryRunExtensionMethod.cs
Comment (pawelbaran) [inline Reflection_Engine/Compute/TryRunExtensionMethod.cs:49]:
```suggestion
        [Input("result", "Result of the method invocation, if the method had been invoked. If no method is found this is null.")]
```

**Review by pawelbaran — COMMENTED** [review]
I have a found minor misconceptions in the descriptions - please see the code comments (plus possibly consider copy-pasting stuff from `RunExtensionMethod`).

On top of that, I would consider two changes in the code itself:
- Adding a similar method taking allowing to call an extension method with other parameters than the target (similar as in `RunExtensionMethod`).
- Refactoring the method to return `Output<bool, object>` instead of `bool` and `out` param - this would allow using the method from the UI level, which may be a nice to have. However, in this case I would lean on @adecler to have the final say.

File: Reflection_Engine/Compute/TryRunExtensionMethod.cs
Comment (pawelbaran) [reply Reflection_Engine/Compute/TryRunExtensionMethod.cs:50]: Ah, BTW the output is missing name.

**Comment by alelom** [comment]
> Refactoring the method to return Output<bool, object> instead of bool and out param - this would allow using the method from the UI level, which may be a nice to have. However, in this case I would lean on @adecler to have the final say.

My preference is to keep the `out` parameter instead, to follow C#'s convention. In particular, I expect this method to be used 95% of the time from the code, rather than from the UI. Also, I believe that the ideal solution for scenarios like these would be for the framework to support the correct reflection of `out` parameters in the outputs of a component (as far as I understand, currently `out` parameters end up among the inputs instead).

**Comment by pawelbaran** [comment]
Agreed with @alelom, the standard C# convention with `out` parameter makes sense to me in this particular case, so the code LGTM now.

However, due to the fact that the PR introduces a UI glitch (nonharmful, the UI is not crashing, it just looks weird), I would like @adecler to have a final say before we approve/merge.

**Review by adecler — APPROVED** [review]
I am fine with the `out` parameter. It better aligns with the general C# conventions on the `Try`  methods and we will eventually support `out` parameters in the UI anyways. In the meantime, people can just use the `RunExtensionMethod` component in the UI if they ever need that for some reason.

**Comment by pawelbaran** [comment]
@BHoMBot check all

**Review by pawelbaran — APPROVED** [review]
Confirming LGTM, happy to see this merged.

**Pull request merged**
